### ISSUE TITLE
Clicks and pops on Scarlett 18i20

### ISSUE THREAD

**Issue by aiac** [Andersama/obs-asio#33]
No matter which buffer size I choose there are many clicks and pops while recording :( Added just two SPDI/F channels to OBS source.

**Comment by pkviet** [comment]
that's odd; not sure how i can help you given the paucity of infos you're giving. Drop an obs log here, maybe i'll spot something.

**Comment by INVLNME** [comment]
I've got the same problem, I tried 1.3.2 after trying 2.0.0 and had the same problem however using 1.3.2 I was able to access the 'ASIO' drop-down menu unlike 2.0.0 and none of the selectable buffer rates matched what the 18i20 could select between e.g. 16,32,48,64,96,128 so on to 1024, OBS could only pick between rates that didn't quite match like 1002 or 46.
I suspect there is a buffer mismatch between everything else running a 512 and obs trying to run a rate slightly off, reading the log  
"18:04:44.366: adding 64 milliseconds of audio buffering, total audio buffering is now 64 milliseconds (source: Mic/Aux)" 
seems weird as 512 samples roughly equate to 10ms but I don't know if this line has anything to do the problem at hand.
Hope this helps as the only work around I found was to run a separate computer or a second interface with noisey inputs...
[asio log.txt](https://github.com/Andersama/obs-asio/files/3994015/asio.log.txt)

**Comment by pkviet** [comment]
1) The ASIO drop down is not necessary anymore since 2.0.0 because the new release is multi-device capable; the previous was not so one had to select the enabled device in case one has several asio sound cards.
2) what you see in the log is unrelated to ASIO; it's with a global audio source (Mic/aux) set in Settings > Audio while ASIO is set exclusively as a separate source.
3) the log you linked is not helpful; i only see that the dll was loaded. Retrieve the log after streaming/recording and exiting obs. To get it, you relaunch obs and search it in the log folder (Help menu > logs). I'll see if obs-asio spouts any warning.
But given there are issues both for 1.3.2 and 2.0.0, my guess is that the issue is not on the plugin side, nor obs.
The buffer might be set by another app. Remember that ASIO is exclusive; you can't ever use a device with two apps (unless the manufacturer departed from ASIO sdk and enabled it, ex: RME). So if your focusrite is used both by OBS and a DAW, you're f...

**Comment by pkviet** [comment]
closing since there's been no movement. Feel free to re-open.

**Comment by armak** [comment]
Just noticed this same issue as well. First generation 18i8, latest drivers.

**Comment by armak** [comment]
It appears some samples are simply dropped and replaced with zeros. This issue seems to be present whenever the ASIO input plugin is active in OBS, both in the recorded input (via SPDI/F) _as well as_ the regular output of the device. Leads me to believe that the issue is not with recording the input, but somehow the plugin not behaving nicely with the Focusrite ASIO driver. Could the plugin be causing bad enough interrupt latency for the driver that the it incurs some buffer underruns?
![image](https://user-images.githubusercontent.com/4991927/88319291-9e811d80-cd24-11ea-9cce-c9d981fd9d3c.png)

Here's also a recording of a sine wave with the crackling extremely apparent (.zip because cannot attach audio data to GitHub comments).
[focusrite-obs-asio.zip](https://github.com/Andersama/obs-asio/files/4967812/focusrite-obs-asio.zip)

**Comment by pkviet** [comment]
What's your latency ?
Try to increase it to play it safe.
As I don't own the device it's very difficult to troubleshoot.
Anything standing out in the logs ?
You might want to try the bassasio version of the plugin.
(But you'll have to compile it. I might setup in the future a way to enable
people to use github actions to compile for them.)

**Comment by saspml** [comment]
I can confirm by recording audio (pcm s24le) at the same time using OBS ASIO and Bitwig ASIO, sample drops only happen with OBS ASIO.

I have reproduced this multiple times.

**Comment by saspml** [comment]
I did another test, using WASAPI instead of ASIO.

I recorded the audio twice at the same time, using OBS & WASAPI and Audacity & WASAPI.

I do not have dropped samples on the Audacity file.

I have only dropped samples on the OBS recording.

It seems I have narrowed is down to OBS being the issue.

There are a lot of users reporting the same issue on the OBS forum https://obsproject.com/forum/threads/audio-dropouts-when-recording.123058/page-2
